### DISCUSSION TITLE
Can't save PDF file after making annotations

### DISCUSSION THREAD

**Discussion by vipatel10** [sumatrapdfreader/sumatrapdf#2778] [Q&A]
I had no trouble saving annotations to this same PDF for a couple months now and all of a sudden I'm getting the error in the screen shot, "Saving of history.pdf failed with "cannot remove file"  history.pdf: Permission denied."  I'm using 3.5.15218.  I checked and the file is not marked 'read only' by windows.  I tried renaming it and moving it to another folder.  I don't have this issue with other PDF files
![sumatra screenshot](https://user-images.githubusercontent.com/4982696/177048527-fc867f34-80b2-4f40-84c5-226a40f988f2.jpg)
.

**Comment by vipatel10** [comment]
Update:  When I try to open the file in Acrobat DC I get the error message: "There was an error opening this document.  The file is damaged and could not be repaired."  Somehow the file opens fine in Sumatra though.  Is there a way to fix the PDF?

**Reply by GitHubRulesOK** [reply]
Annotation should be added as a save on closure, preferably not repeatedly since  each rewrite is a potential for a file mutation. The annotation is added "incrementally" each time it is saved.
![image](https://user-images.githubusercontent.com/12754913/177052510-1d30224c-41d7-4206-86a3-6c7065392acf.png)

There was a good freeware tool to remove the lase written data thus remove any single bad last session, but that tool is no longer available. 

Since SumatraPDF is based on MuPDF the simplest way to try similar is open the file in MuPDF-GL which should show any simple repair by means of * in the heading, then if viable press A to resave annotations. I find it difficult to make a bad file for illustration so can only say that on saving the file will likely be recovered but any fails in annotation will likely be deleted, but may not be the case.
![image](https://user-images.githubusercontent.com/12754913/177052941-44458890-612d-49ab-bb22-af8d638a7c78.png)

If you can upload a file I can test for which recovery point is possible, but the easiest way to hand correct is with a copy use a "Hex" editor to find Last good %%EOF (may be the one before last) then delete everything after. so for this example here before the highlighted area is %%EOF with x0A (linefeed) nThen the highlight shows me looking at suspect annotation so that selection and everything else afterwards needs deleting to restore the file to before that last annotation. There are other methods but again without an example I could not guess why or what needs adjusting.

![image](https://user-images.githubusercontent.com/12754913/177053409-38e6308c-17aa-4d94-a317-ef1ec78f2fe7.png)
